Workade/Extentions/UIImage+.swift

```diff
@@ -8,7 +8,40 @@
 import UIKit
 
 extension UIImage {
+    static func fromSystemImage(name: String, font: UIFont, color: UIColor = .black) -> UIImage? {
+        let configuration = UIImage.SymbolConfiguration(font: font)
+        var image = UIImage(systemName: name, withConfiguration: configuration)
+        image = image?.withTintColor(color, renderingMode: .alwaysOriginal)
+        return image
+    }
+
```

Comment:
[P3] static으로 선언하는게 과연 맞는 방식인가에 대해서는 조금 의문이 듭니다. 메모리적인 측면이 조금 신경쓰입니다.

Reply:
```suggestion
/// 이 주석에 대한 설명
    static func fromSystemImage(name: String, font: UIFont, color: UIColor = .black) -> UIImage? {
        let configuration = UIImage.SymbolConfiguration(font: font)
        var image = UIImage(systemName: name, withConfiguration: configuration)
        image = image?.withTintColor(color, renderingMode: .alwaysOriginal)
        return image
    }

```
[P3] 주석을 활용했으면 하는 방안입니다. 코드 리뷰하는 입장에서는 이 함수가 왜 필요하지? 라는 생각이 드는데 꽤 많은 시간이 듭니다. /// 주석을 통해서 코드리뷰하는 사람의 가독성을 올리면 좋을 것 같습니다. 위 주석을 활용하면 아래와 같이 한 눈에 볼 수 있습니다.
![스크린샷 2022-10-25 오전 1 54 40](https://user-images.githubusercontent.com/55151796/197582266-d7a1b34a-dbdf-46f6-9bcd-53147ffd740d.png)

Reply:
[P3] 우선 static을 쓴 이유에 대해 먼저 말씀드릴게요! 평소 저는 이미 생성된 인스턴스의 속성값을 바꾸는 것이 목적이라면 인스턴스 메서드로 만들고, 기존 인스턴스와의 관계없이 해당 타입과 관련된 새로운 것을 찍어내는 느낌이라면 타입 메서드를 사용합니당. 저 메서드를 설계할 때, 처음엔 UIImage(systemName: "bookmark") <- 이렇게 생성된 인스턴스의 뒤에 붙는 인스턴스 메서드로 설계하고자했으나, 위의 withTintColor나 withConfiguration 메서드의 경우 image = image.withTintColor(...) <- 이런 형태로 반환값을 이용하여 변경하는 메서드입니당. 하지만 인스턴스 메서드 속에서 self = self.withTintColor(...) <- self는 불변값이니 최종적으로 static하게 설계했습니당!

Reply:
[P3] 와우 /// 주석 예~전에 들었던 것 같은데 잊고있었네용 ㅠㅠ 오 저렇게하면 너무 좋을 것 같습니다! 참고해서 저렇게 커스텀한 요소들에게 설명 주석 붙이도록 하겠습니다~! 굳굳

Reply:
아하 이해했습니다. `UIImage(systemName:).fromSystemImage()`의 형태를 피하고자 하는 것도 있군요.
그럼 나중에 저 함수가 많이 쓰일 것 같은데 함수 정의를 해주면 좋을 것 같습니다.
[퀵 헬프 마크업 문법](https://charmintern.tistory.com/41) 참고하시면 좋은 사이트를 같이 첨부합니다!